src/ds/d500/d500-device.cpp

```diff
@@ -0,0 +1,838 @@
+        register_info(RS2_CAMERA_INFO_FIRMWARE_UPDATE_ID, asic_serial);
+        register_info(RS2_CAMERA_INFO_FIRMWARE_VERSION, gvd_parsed_fields.fw_version);
+        register_info(RS2_CAMERA_INFO_PHYSICAL_PORT, group.uvc_devices.front().device_path);
+        register_info(RS2_CAMERA_INFO_DEBUG_OP_CODE, std::to_string(static_cast<int>(fw_cmd::GLD)));
+        register_info(RS2_CAMERA_INFO_ADVANCED_MODE, ((advanced_mode) ? "YES" : "NO"));
+        register_info(RS2_CAMERA_INFO_PRODUCT_ID, pid_hex_str);
+        register_info(RS2_CAMERA_INFO_PRODUCT_LINE, "D500");
+        // Uncomment once D500 recommended FW exist
+        //register_info(RS2_CAMERA_INFO_RECOMMENDED_FIRMWARE_VERSION, _recommended_fw_version);
+        register_info(RS2_CAMERA_INFO_CAMERA_LOCKED, _is_locked ? "YES" : "NO");
+
+        if (usb_modality)
+            register_info(RS2_CAMERA_INFO_USB_TYPE_DESCRIPTOR, usb_type_str);
+    }
+
+    void d500_device::create_snapshot(std::shared_ptr<debug_interface>& snapshot) const
+    {
+        //TODO: Implement
+    }
+    void d500_device::enable_recording(std::function<void(const debug_interface&)> record_action)
+    {
+        //TODO: Implement
+    }
+
+    platform::usb_spec d500_device::get_usb_spec() const
+    {
+        if(!supports_info(RS2_CAMERA_INFO_USB_TYPE_DESCRIPTOR))
+            return platform::usb_undefined;
+        auto str = get_info(RS2_CAMERA_INFO_USB_TYPE_DESCRIPTOR);
+        for (auto u : platform::usb_spec_names)
+        {
+            if (u.second.compare(str) == 0)
+                return u.first;
+        }
+        return platform::usb_undefined;
+    }
+
+
+    double d500_device::get_device_time_ms()
+    {
+        //// TODO: Refactor the following query with an extension.
+        //if (dynamic_cast<const platform::playback_backend*>(&(get_context()->get_backend())) != nullptr)
+        //{
+        //    throw not_implemented_exception("device time not supported for backend.");
+        //}
+
+        if (!_hw_monitor)
+            throw wrong_api_call_sequence_exception("_hw_monitor is not initialized yet");
+
+        command cmd(ds::MRD, ds::REGISTER_CLOCK_0, ds::REGISTER_CLOCK_0 + 4);
+        auto res = _hw_monitor->send(cmd);
+
+        if (res.size() < sizeof(uint32_t))
+        {
+            LOG_DEBUG("size(res):" << res.size());
+            throw std::runtime_error("Not enough bytes returned from the firmware!");
+        }
+        uint32_t dt = *(uint32_t*)res.data();
+        double ts = dt * TIMESTAMP_USEC_TO_MSEC;
+        return ts;
+    }
+
+    command d500_device::get_firmware_logs_command() const
+    {
+        return command{ ds::GLD, 0x1f4 };
+    }
+
+    command d500_device::get_flash_logs_command() const
+    {
+        return command{ ds::FRB, 0x17a000, 0x3f8 };
+    }
+
+    bool d500_device::check_symmetrization_enabled() const
+    {
+        command cmd{ ds::MRD, 0x80000004, 0x80000008 };
+        auto res = _hw_monitor->send(cmd);
```

Comment:
@maloel need to verify if needed for this project

Reply:
What does it mean?

Reply:
Do we have a list of changes D500-vs-D400?

Reply:
No.. 
But we also need D5XX products diff.. this is why I place my comments, so you should verify about those topics.

Reply:
This is a command that checks if the camera in running symmetrization algo as part of the depth pipe.
It affects some calculations in the SDK , could be relevant, could be not